src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Predicates/PredicateQuery.cs

```diff
@@ -60,8 +62,17 @@ public string GetColumnName(string propertyPath)
                 {
                     // Return the default alias
                     // ContentItemId -> ContentItemIndex.ContentItemId
-                    _usedAliases.Add(alias);
-                    return Dialect.QuoteForTableName($"{_tablePrefix}{alias}") + "." + Dialect.QuoteForColumnName(values[0]);
+                    // Ensures its actually a property of ContentItemIndex and gets the correct casing.
+                    var propertyName = values[0];
+
+                    var contentItemIndexProperty = typeof(ContentItemIndex)
```

Comment:
Please use a local cache (static concurrent dictionary per type + property name) to prevent reflection